pkg/apis/core/types.go

```diff
@@ -5788,3 +5796,470 @@ type PortStatus struct {
+	// allocated.
+	//
+	// The ResourceClass.SuiteableNodes node selector can be used to filter
+	// out nodes based on labels. This prevents adding nodes here that the
+	// driver then would need to reject through UnsuitableNodes.
+	//
+	// The size of this field is limited to 256 (=
+	// [PodSchedulingNodeListMaxSize]).  This is large enough for many
+	// clusters. Larger clusters may need more attempts to find a node that
+	// suits all pending resources.
+	PotentialNodes []string
+}
+
+// PodSchedulingStatus is where resource drivers provide information about
+// where the could allocate a resource and whether allocation failed.
+type PodSchedulingStatus struct {
+	// Each resource driver is responsible for providing information about
+	// those resources in the Pod that the driver manages. It can skip
+	// adding that information when it already allocated the resource.
+	//
+	// A driver must add entries here for all its pending claims, even if
+	// the ResourceSchedulingStatus.UnsuitabeNodes field is empty, because
+	// the scheduler may decide to wait with selecting a node until it has
+	// information from all drivers.
+	Claims []ResourceClaimSchedulingStatus
+
+	// If there ever is a need to support other kinds of resources
+	// than ResourceClaim, then new fields could get added here
+	// for those other resources.
+}
+
+// ResourceClaimSchedulingStatus contains information about one particular
+// claim while scheduling a pod.
+type ResourceClaimSchedulingStatus struct {
+	// PodResourceClaimName matches the PodResourceClaim.Name field.
+	PodResourceClaimName string
+
+	// UnsuitableNodes lists nodes that the claim cannot be allocated for.
+	// Nodes listed here will be ignored by the scheduler when selecting a
+	// node for a Pod. All other nodes are potential candidates, either
+	// because no information is available yet or because allocation might
+	// succeed.
+	//
+	// A change of the PodSchedulingSpec.PotentialNodes field and/or a failed
+	// allocation attempt trigger an update of this field: the driver
+	// then checks all nodes listed in PotentialNodes and UnsuitableNodes
+	// and updates UnsuitableNodes.
+	//
+	// It must include the prior UnsuitableNodes in this check because the
+	// scheduler will not list those again in PotentialNodes but they might
+	// still be unsuitable.
+	//
+	// This can change, so the driver also must refresh this information
+	// periodically and/or after changing resource allocation for some
+	// other ResourceClaim until a node gets selected by the scheduler.
+	//
+	// The size of this field is limited to 256 (=
+	// [PodSchedulingNodeListMaxSize]), the same as for
+	// PodSchedulingSpec.PotentialNodes.
+	UnsuitableNodes []string
+}
+
+// PodSchedulingNodeListMaxSize defines the maximum number of entries in the
+// node lists that are stored in PodScheduling objects. This limit is part
+// of the API.
+const PodSchedulingNodeListMaxSize = 256
```

Comment:
can we inform this better? How many bytes does this limit imply?

Reply:
256 * maximum size of a node name, so 256 * 253 ("DNS1123SubdomainMaxLength is a subdomain's max length in DNS (RFC 1123)") -> ~64KiB.

[According to Tim](https://github.com/pohly/enhancements/pull/13#issuecomment-1163400602), this can be large, it just must not be unlimited.

Reply:
Quite a bit already. We capped the limits at 20KiB for other KEPs. Maybe we should do 128.

Reply:
Okay, let's do 128 for now. Changed.